TrainingGym/train.py

```diff
@@ -11,58 +11,92 @@
+            proc.join()
+        log_to_file(f'    Done trial {i_trial} in {time.time() - trial_start_time} s')
+    log_to_file(f'FINISHED TRAINING LOOP in {time.time() - loop_start_time} s')
 
-        # Create an environment where the ith light in controlled_lights is being trained
-        env = create_env(learning_light['name'], num_proc, steps_per_episode)
 
-        # Load existing model for the learning light if it exists
-        path_name = f'Scenarios/{config_params["model_save_path"]}/PPO2_{learning_light["name"]}'
-        if os.path.isfile(path_name + '.zip'):
-            model = PPO2.load(path_name, env=env)
+def learn_light(learning_light, num_workers, global_config_params, config_params):
+
+    # Create an environment where the ith light in controlled_lights is being trained
+    env = create_env(learning_light['light_name'], num_workers, config_params['steps_per_episode'],
+                     global_config_params['visualize_training'])
+
+    # Load existing model for the learning light if it exists
+    model_path = f'Scenarios/{config_params["model_save_path"]}/PPO2_{learning_light["light_name"]}'
+    if global_config_params['use_tensorboard']:
+        if os.path.isfile(model_path + '.zip'):
+            model = PPO2.load(model_path, env=env,
+                              tensorboard_log=f'./Scenarios/{config_params["model_save_path"]}/tensorboard/{learning_light["light_name"]}/')
+        else:
+            model = PPO2(MlpPolicy, env, verbose=1,
+                         tensorboard_log=f'./Scenarios/{config_params["model_save_path"]}/tensorboard/{learning_light["light_name"]}/')
+    else:
+        if os.path.isfile(model_path + '.zip'):
+            model = PPO2.load(model_path, env=env)
         else:
             model = PPO2(MlpPolicy, env, verbose=1)
 
-
-        train_start_time = time.time()
-        model.learn(total_timesteps=steps_per_episode * num_episodes)
-        print(f'LEARNING TIME: {time.time() - train_start_time}')
-        model.save(f'Scenarios/{config_params["model_save_path"]}/PPO2_{learning_light["name"]}')
-        print(f'DONE LEARNING LIGHT: {learning_light["name"]}')
-        env.close()
-        del env
+    train_start_time = time.time()
+    model.learn(total_timesteps=config_params['steps_per_episode'] * config_params['num_episodes'])
+    print(f'LIGHT LEARNING TIME: {time.time() - train_start_time}')
+    model.save(f'Scenarios/{config_params["model_save_path"]}/PPO2_{learning_light["light_name"]}')
+    print(f'DONE LEARNING LIGHT: {learning_light["light_name"]}')
+    env.close()
+    del env
 
 
 # Create a sumo environment
-def create_env(node_name, num_proc, steps_per_episode):
+def create_env(node_name, num_proc, steps_per_episode, visualize_training):
     if num_proc == 1:
-        env = DummyVecEnv([lambda: SumoEnvParallel(steps_per_episode, False, node_name)])
+        env = DummyVecEnv([lambda: SumoEnvParallel(steps_per_episode, visualize_training, node_name)])
```

Comment:
we can just return right here

`return  DummyVecEnv([lambda: SumoEnvParallel(steps_per_episode, visualize_training, node_name)])`

This has the added bonus of not overwriting our `env` library variable